client/client.go

```diff
@@ -37,21 +41,97 @@ func New(logID int64, cc *grpc.ClientConn) *LogClient {
 	}
 }
 
-// AddLeaf adds leaf to the append only log. It blocks until a verifiable response is received.
+// AddLeaf adds leaf to the append only log.
+// Blocks until it gets a verifiable response.
 func (c *LogClient) AddLeaf(ctx context.Context, data []byte) error {
+	leaf := buildLeaf(data)
+	if err := c.queueLeaf(ctx, leaf); err != nil {
+		return err
+	}
+
+	// Wait for TreeSize to update.
+	if err := c.waitForSTRUpdate(ctx, 2); err != nil {
```

Comment:
This won't block needlessly because it uses current STR in the LogClient as it's base line rather than the first fetch. I've added an `UpdateSTR` call before queue leaf